src/genetic_low_memory.c

```diff
@@ -0,0 +1,460 @@
+
+  if (mutation <= mutationRate) {
+
+    const unsigned int mutatedIndex =
+        linear_congruential_random_generator() * (geneLength - 1);
+
+    // We then generate a number between 0 and 9 to replace the chosen digit
+    unsigned int newValue = linear_congruential_random_generator() * 9;
+
+    // We handle the last digit
+    if (mutatedIndex % UINT16_DIGIT_COUNT == 0) {
+      newValue = linear_congruential_random_generator() * 6;
+    }
+    // Converts int to char
+    gene[mutatedIndex] = intDigitToChar(newValue);
+  }
+}
+
+/**
+ * This function divides the created child array in the respective parameters of
+ * the solution before adding the created child solution to the next generation
+ *
+ * @param nextGeneration  the array containing all of the created chilren
+ * @param nextGenerationSize the amount of values in the array of created
+ * children
+ * @param child the encoded string of the new child
+ */
+static void decodeAndAddChild(uint16_t* nextGeneration,
+                              unsigned int* nextGenerationSize, char* child) {
+
+  unsigned int dimensionIndex = 0;
+
+  char parameter[UINT16_DIGIT_COUNT + 1];
+  parameter[UINT16_DIGIT_COUNT] = '\0';
+
+  unsigned int basePositionning = *nextGenerationSize * dimensions;
+
+  for (unsigned int i = 0; i < dimensions; i++) {
+
+    for (unsigned int j = 0; j < UINT16_DIGIT_COUNT; j++) {
+
+      parameter[j] = child[j + dimensionIndex];
+    }
+
+    dimensionIndex += UINT16_DIGIT_COUNT;
+
+    uint32_t value = atoi(parameter);
+
+    if (value > UINT16_MAX) {
+      value = UINT16_MAX;
+    }
+
+    nextGeneration[basePositionning + i] = value;
+  }
+  (*nextGenerationSize)++;
+  basePositionning++;
+}
+
+/**
+ *
+ * We create the children based off of a uniform crossover with two parents and
+ * apply the genetic operators
+ *
+ * @param firstParent  the first selected parent solution
+ * @param secondParent  the second selected parent solution
+ * @param nextGeneration the array containing all of the created chilren
+ * @param nextGenerationSize the amount of values in the array of created
+ * children
+ */
+static void createChildren(char* firstParent, char* secondParent,
+                           uint16_t* nextGeneration,
+                           unsigned int* nextGenerationSize) {
+
+  const unsigned int arraySize = UINT16_DIGIT_COUNT * dimensions;
+  const unsigned int encodedChildStringSize = arraySize + 1;
+
+  char firstChildString[encodedChildStringSize];
+  char secondChildString[encodedChildStringSize];
+
+  firstChildString[arraySize] = '\0';
+  secondChildString[arraySize] = '\0';
+
+  for (unsigned int i = 0; i < arraySize; i++) {
+
+    if (linear_congruential_random_generator() <= 0.5) {
+
+      firstChildString[i] = firstParent[i];
+      secondChildString[i] = secondParent[i];
+    } else {
+      firstChildString[i] = secondParent[i];
+      secondChildString[i] = firstParent[i];
+    }
+  }
```

Comment:
Instead of the for loop, you could do that, I think it's more efficient : 
```
if (linear_congruential_random_generator() <= 0.5) {
    memcpy(firstChildString, firstParent, arraySize);
    memcpy(secondChildString, secondParent, arraySize);
} else {
    memcpy(firstChildStrign, secondParent, arraySize);
    memcpy(secondChildString, firstParent, arraySize);
}
```

Reply:
I will look into it to see if its more efficient

Reply:
I ran both in godbolt and you are right its much more efficient.
https://godbolt.org/z/YYEfhEv8Y
https://godbolt.org/z/5WY7da1aK